### ISSUE TITLE
Supporting Immutable.JS in `serializable-state-invariant-middleware`

### ISSUE THREAD

**Issue by kgregory** [reduxjs/redux-starter-kit#140]
I really like the problem that `serializable-state-invariant-middleware` is meant to solve, but I use `immutable` structures in my store, which will always be seen as non-serializable by the current version of this middleware.  

I think a minor change would allow me to augment this middleware to allow the use of `immutable`.

Add a new option that will retrieve the entries for a given value:

```js
export interface SerializableStateInvariantMiddlewareOptions {
  /**
   * The function to check if a value is considered serializable. This
   * function is applied recursively to every value contained in the
   * state. Defaults to `isPlain()`.
   */
  isSerializable?: (value: any) => boolean,
  /**
   * The function that will be used to retrieve entries from each
   * value.  If unspecified, `Object.entries` will be used. Defaults
   * to `undefined`.
   */
  getEntries?: (value: any) => [string, any][],
}
```

Consumers that use `immutable` can write their own functions:

```js

// immutable version 3.8.2 assumed for this demo!
const isSerializable = (value: any): boolean =>
  Iterable.isIterable(value) || isPlain(value);

const getEntries = (value: any): [string, any][] =>
  Iterable.isIterable(value) ? value.entries() : Object.entries(value);

const middleware = createSerializableStateInvariantMiddleware({
  isSerializable,
  getEntries,
});
```

We can accept these functions as part of the options provided at creation:

```js
export function createSerializableStateInvariantMiddleware(
  options: SerializableStateInvariantMiddlewareOptions = {}
): Middleware {
  const { isSerializable = isPlain, getEntries } = options

  return storeAPI => next => action => {
    const foundActionNonSerializableValue = findNonSerializableValue(
      action,
      [],
      isSerializable,
      getEntries,
    )
    // ...
```

We can use `getEntries` in `findNonSerializableValue` (if provided):

```js
export function findNonSerializableValue(
  value: unknown,
  path: ReadonlyArray<string> = [],
  isSerializable: (value: unknown) => boolean = isPlain,
  getEntries?: (value: unknown) => [string, any][]
): NonSerializableValue | false {
  // ...
```

Switch to `Object.entries` when evaluating nested values:

```js
  let entries;
  if (getEntries != null) {
    entries = getEntries(value);
  } else {
    entries = Object.entries(value);
  }

  for (const [property, nestedValue] of entries) {
    const nestedPath = path.concat(property)
    // ...
```

This will skip anything that is an `immutable` structure and error for anything else.

I didn't want to submit an unwanted PR, so please let me know if you think this would be a welcome submission.  Thanks for your time and attention!  This is a great library.

**Comment by kgregory** [comment]
>I didn't want to submit an unwanted PR

I was working on it and figured the PR can be scrapped if it isn't wanted.

**Comment by markerikson** [comment]
Should be resolved now that #141 is merged.  I'll try to put out a release once I've done some work on our build system.
